### PR TITLE
experiment bug fix

### PR DESCRIPTION
**Pull request by TeaganKing**
This PR involves some changes to the arguments in run_case in order to address #2433

### CODE REVIEW

**Comment by wwieder** [comment]
Assigning this to @slevis-lmwg to integrate into b4b-dev after @ekluzek is able to approve the PR.

**Comment by wwieder** [comment]
I should add that I was able to create --transient cases with code mods on this branch.

**Comment by wwieder** [comment]
@TeaganKing I also see this is marked as a WIP.  Was there more you were planning to add to this PR?

File: python/ctsm/site_and_regional/neon_site.py [hunk 1]
```diff
@@ -47,8 +47,6 @@ def build_base_case(
             user_mods_dirs = [
                 os.path.join(self.cesmroot, "cime_config", "usermods_dirs", "NEON", self.name)
             ]
-            print("in neonsite adding usermodsdirs")
```
Comment (ekluzek) [inline python/ctsm/site_and_regional/neon_site.py:50]: Looks like removing some debugging prints. Makes sense...

File: python/ctsm/site_and_regional/neon_site.py [hunk 2]
```diff
@@ -99,8 +97,20 @@ def run_case(
             os.path.join(self.cesmroot, "cime_config", "usermods_dirs", "NEON", self.name)
         ]
         tower_type = "NEON"
+
         super().run_case(
-            base_case_root, run_type, prism, run_length, user_version, tower_type, user_mods_dirs
+            base_case_root,
```
Comment (ekluzek) [inline python/ctsm/site_and_regional/neon_site.py:102]: I'm guessing this is just a formatting change that running through black did...

File: python/ctsm/site_and_regional/tower_site.py [hunk 3]
```diff
@@ -257,11 +257,11 @@ def run_case(
         user_version,
         tower_type,
         user_mods_dirs,
-        overwrite=False,
-        setup_only=False,
-        no_batch=False,
-        rerun=False,
-        experiment=False,
+        overwrite,
```
Comment (ekluzek) [inline python/ctsm/site_and_regional/tower_site.py:260]: So this is changing the defaults that allow these options to NOT to be set, to be required to be set. I suppose if these options are always provided there's no reason to have the default. But, I wasn't sure why this change was being made?

**Review by ekluzek — APPROVED** [review]
Marking as approved since it's small and I see no problems with the changes.

This is still marked as a draft, is this ready to come in? Or is there more going on here?

I wasn't sure about the removal of the default values from run_case. Can you elaborate more on why that's being done?

**Comment by TeaganKing** [comment]
@wwieder I was planning to add fixes for other issues in the same PR, but I think maybe we should keep this as simple as possible since there are lots of moving pieces right now. I wasn't planning to add anything else specifically for the 'experiment' functionality.

File: python/ctsm/site_and_regional/tower_site.py (hunk 3 above)
Comment (TeaganKing) [reply python/ctsm/site_and_regional/tower_site.py:260]: These values are not always set by the user. I was noticing that tower_site was not correctly pulling information from neon_site if neon_site didn't set all of the parameters. If there's a better way to do this, I'm all ears :)

Comment (TeaganKing) [reply python/ctsm/site_and_regional/tower_site.py:260]: Instead of the default parameter being set in neon_site, it's now in tower_site.

Comment (TeaganKing) [reply python/ctsm/site_and_regional/tower_site.py:260]: Instead of the default parameter being set in neon_site, it's now in tower_site.

**Comment by slevis-lmwg** [comment]
Testing after merging b4b-neon to this branch (thus, effectively, having merged the latest b4b-dev):
OK: ./run_ctsm_py_tests -u
FAIL: ./run_ctsm_py_tests -s
IN PROGRESS: ./run_sys_tests -s clm_pymods -c ctsm5.1.dev176 --skip-generate

@TeaganKing two requests:
1) Could you add me as collaborator so that I can push to this PR? You would click on Settings just to the right of middle-top of this page. Then click on Collaborators and follow the instructions for adding me (@slevis-lmwg).
2) The python system testing triggers this error:
```
ERROR: test_one_site (test.test_sys_run_neon.TestSysRunNeon)
This test specifies a site to run
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/glade/work/slevis/git/latest_master/python/ctsm/test/test_sys_run_neon.py", line 57, in test_one_site
    main("")
  File "/glade/work/slevis/git/latest_master/python/ctsm/site_and_regional/run_neon.py", line 242, in main
    experiment=experiment,
  File "/glade/work/slevis/git/latest_master/python/ctsm/site_and_regional/neon_site.py", line 113, in run_case
    experiment,
  File "/glade/work/slevis/git/latest_master/python/ctsm/site_and_regional/tower_site.py", line 305, in run_case
    self.name = self.name + "." + experiment
TypeError: can only concatenate str (not "NoneType") to str
```
I think I should send this back to you to fix, unless @wwieder would rather I did the rest of the debugging on this.

**Comment by TeaganKing** [comment]
Thanks for catching that, @slevis-lmwg . I added you to the repository. I'm not going to be able to fix the system test error tonight, but I think I see what's going on and should have time to fix it tomorrow.

**Comment by slevis-lmwg** [comment]
Thank you, @TeaganKing

**Comment by wwieder** [comment]
@TeaganKing is this something that's straight forward for you to address.

**Comment by TeaganKing** [comment]
Hi @slevis-lmwg, I think I fixed the issue.

**Comment by slevis-lmwg** [comment]
For my own sanity, I'm also running
`./run_sys_tests -s aux_clm -c ctsm5.1.dev176 --skip-generate`
I didn't do so for #2472 before merging it to b4b-dev, so I'm doing it now and hoping for the best...
